File: Questions/Java/Chef_and_party.java
import java.util.*;

class Selection_sort
{
	public static void main(String[] args)
	{
		Scanner sc = new Scanner(System.in);
		/* code */
		int t= sc.nextInt();	//Input length of array = n 
    for(int k=0;k<t;k++)
    {
       int n= sc.nextInt();
       int[] v=new int[n];

       for(int i=0;i<n;i++)
       {
         v[i]= sc.nextInt();
       }

       Arrays.sort(v);

       int count=0;
       for(int i=0;i<n;i++)
       {

            if(v[i]<=count)
            {
                count++;
            }
            else
            {
                break;
            }
       }

       System.out.println(count);
    }
  }
}

File: Questions/chef_and_party.cpp
#include<bits/stdc++.h>

using namespace std;

int main()
{
    int t;
    cin>>t;
    for(int k=0;k<t;k++)
    {
       int n;
       cin>>n;
       std::vector<int> v(n,0);
       int inp;

       for(int i=0;i<n;i++)
       {
         cin>>v[i];
       }

       sort(v.begin(),v.end());

       int count=0;
       for(int i=0;i<n;i++)
       {

            if(v[i]<=count)
            {
                count++;
            }
            else
            {
                break;
            }
       }

       cout<<count<<endl;
    }

    return 0;
  }